New optional parameter to specify node version

rsperre opened the pull request:
With SPFx 1.9.1 build and test fails due to node 12.x not being supported by node-sass@4.9.3 that is required by @microsoft/gulp-core-build-sass.

Being able to specify to run against node version 10.x allows these jobs to complete successfully.

andrewconnell commented:
I see these submissions but none of the builds are passing. Will require me to take a deeper look at what you've submitted. Just letting you know I'm not ignoring it, just a bit swamped with higher priorities.

rsperre commented:
No worries - The reason the tests fail is actually what this PR fixes. It needs to run against node v 10.x Not sure if the pipeline actually uses the yaml included in the sample?

andrewconnell commented:
Yes... it uses what's in the example project which uses the templates.

rsperre commented:
Strange. It is not executing the task I added:
` - task: UseNode@1`

andrewconnell commented:
Looks like you're still pointing to the upstream repo... see this: https://github.com/Voitanos/azure-pipelines-spfx-templates#important-consider-referencing-specific-template-versions

rsperre commented:
I meant that your pipeline is not executing the task I added. In my own pipeline this is executed right before the NPM step:

![image](https://user-images.githubusercontent.com/12030017/70899093-3df36a80-1ff6-11ea-8770-8ad72e147270.png)

I updated the file `azure-pipelines.yml` in the sample folder to also run against node 10.x, but it doesn't seem like your pipeline is using this configuration directly.

andrewconnell commented:
That's because the DevOps project linked to this repo has a service connection that points to this repo which will pull from it's `master` branch. The changes exist in YOUR fork, so when the pipeline runs, it won't pull the template from your fork, it will pull from this repo.

it will just take time for me to review what you've submitted. Swamped, with end-of-the-year work so please be patient.

andrewconnell commented:
*FINALLY* getting around to updating the repo. Thanks for this PR. I made a few additional changes, but it's now live in the **master** branch as v1.1.
